File: LC1431.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& nums, int ex) {
        vector<bool> v (nums.size(), false);

        int m = 0;

        for(int i = 0; i < nums.size(); i++)
            m = max(m, nums[i]);
        
        for(int i = 0; i < nums.size(); i++)
            if(nums[i] + ex >= m)
                v[i] = true;
        
        return v;
    }
};

File: LC1431.py
class Solution:
    def kidsWithCandies(self, nums: List[int], ex: int) -> List[bool]:

        m = 0
        l = []

        for i in nums:
            m = max(m, i)

        for i in nums:
            if i + ex >= m:
                l.append(True)
            else:
                l.append(False)
        
        return l
        
